module/x/gravity/keeper/signer_set.go

```diff
@@ -0,0 +1,32 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/peggyjv/gravity-bridge/module/v3/x/gravity/types"
+)
+
+// TODO here we should check the contents of the validator set against
+// the store, if they differ we should take some action to indicate to the
+// user that bridge highjacking has occurred
+func (k Keeper) signerSetExecuted(ctx sdk.Context, nonce uint64) {
+	otx := k.GetOutgoingTx(ctx, types.MakeSignerSetTxKey(nonce))
+	if otx == nil {
+		k.Logger(ctx).Error("Failed to clean signer sets",
+			"signer set nonce", nonce)
+		return
+	}
+
+	completedSignerSetTx, _ := otx.(*types.SignerSetTx)
```

Comment:
Should the cast here has similar panic checks as others do?

Reply:
Can I get an example? This is the same thing that `batchExecuted` and `contractCallExecuted` have been doing.

Reply:
```
	signerSet, ok := cotx.(*types.SignerSetTx)
	if !ok {
		panic(sdkerrors.Wrapf(types.ErrInvalid, "couldn't cast to signer set for completed tx %s", cotx))
	}
```